services/storehost/replicationJobRunner.go

```diff
@@ -224,6 +234,19 @@ func (runner *replicationJobRunner) run() {
 		}).Info(`replication for extent started`)
 	}
 
+	var maxConsecutiveFailures int
+	for existingFailedJob, failedTimes := range runner.failedJobs {
+		if _, ok := currentFailedJobs[existingFailedJob]; !ok {
```

Comment:
If the job is missing in the currentFailedJobs, is there any guarantee that the job actually was started / finished from the time we last checked this map ?